src/verkle/mod.rs

```diff
@@ -0,0 +1,297 @@
+
+use ark_serialize::CanonicalDeserialize;
+
+// https://github.com/gballet/verkle-block-sample
+#[allow(dead_code)]
+pub struct VerkleHeader {
+    parent_hash: Vec<u8>,
+    pub storage_root: Vec<u8>,
+    number: Vec<u8>,
+    proof: proof::Proof,
+    keyvals: keyvals::KeyVals,
+}
+
+impl Decodable for VerkleHeader {
+    fn decode(rlp: &rlp::Rlp<'_>) -> Result<Self, rlp::DecoderError> {
+        Ok(VerkleHeader {
+            parent_hash: rlp.at(0)?.as_val::<Vec<u8>>()?,
+            storage_root: rlp.at(3)?.as_val::<Vec<u8>>()?,
+            number: rlp.at(8)?.as_val::<Vec<u8>>()?,
+            proof: rlp.at(16)?.as_val::<proof::Proof>()?,
+            keyvals: rlp.at(17)?.as_val::<keyvals::KeyVals>()?,
+        })
+    }
+}
+
+pub struct VerkleBlock {
+    pub header: VerkleHeader,
+}
+
+impl Decodable for VerkleBlock {
+    fn decode(rlp: &Rlp<'_>) -> Result<Self, DecoderError> {
+        let headerrlp = rlp.at(0)?;
+        let header: VerkleHeader = VerkleHeader::decode(&headerrlp)?;
+        Ok(VerkleBlock { header })
+    }
+}
+
+// with 0x
+fn decode_hex(s: &str) -> Result<Vec<u8>, ParseIntError> {
+    (2..s.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
+        .collect()
+}
+
+fn construct_headers() -> HeaderMap {
+    let mut headers = HeaderMap::new();
+    headers.insert(USER_AGENT, HeaderValue::from_static("test optimal batch size1"));
+    headers.insert(CONTENT_TYPE, HeaderValue::from_static("application/json; charset=utf-8"));
+    headers
+}
+
+// ! rlp string starts with 0x
+pub fn decode_block(rlp: String) -> Result<VerkleBlock, DecoderError> {
+    let serialized: Vec<u8> = decode_hex(&rlp).unwrap();
+    let block: VerkleBlock = decode(&serialized)?;
+
+    Ok(block)
+}
+
+pub fn print_block_info(block: &VerkleBlock) {
+    println!(
+            "Block info:\n- parent hash: {}\n- storage root: {}\n- block number: {}\n",
+            hex::encode(block.header.parent_hash.clone()),
+            hex::encode(block.header.storage_root.clone()),
+            hex::encode(block.header.number.clone())
+        );
+    let keys = block.header.keyvals.keys.clone();
+    println!("Kkey-vals: ");
+    for (indx, key) in keys.iter().enumerate() {
+        match block.header.keyvals.values[indx] {
+            Some(ref val) => println!("\t{} => {}", hex::encode(key), hex::encode(val)),
+            None => println!("\t{} is absent", hex::encode(key)),
+        }
+    }
+}
+
+pub async fn get_rlp(block_number: u64) -> Result<String, reqwest::Error> {
+    let client = Client::new();
+    let node_end_point: String = "https://rpc.condrieu.ethdevops.io/".to_owned();
+    let arg = format!(r#"{{"jsonrpc":"2.0","method":"debug_getBlockRlp","params":[{}],"id":"1"}}"#, block_number);
+
+    let res = client.post(&node_end_point)
+        .body(arg)
+        .headers(construct_headers())
+        .send()
+        .await?;
+
+    let block_rlp: ResponseTest = res.json().await?;
```

Comment:
is this a testing function or ??
why the response called `Test`? what does it test?..